### DISCUSSION TITLE
Box VS Convex is not accurate as Capsule VS convex in scene->overlap or trigger

### DISCUSSION THREAD

**Discussion by yananZZZZ** [NVIDIA-Omniverse/PhysX#174] [Q&A] [answered]
Hi,
     What's I am doing is I create a rather small cylinder like convex shape geometry and make it as overlap query shape. The radius is 0.45mm and 12mm for height.
     As i use a box collision shape (1mm for extend) to do the overlap test, the overlap is not report anything when the box already enter the convex shape. I did the same test only change the collision shape into capsule(1mm for diameter). This time the overlap can get the result as soon as the capsule enter the volume.
     So i look into the code and found out for Box vs Convex a "RelativeConvex" is used for GJK and for capsule is "LocalConvex". I just wonder is this cause the result different? I also found out for small size convex, actually the GJK algorithm is easy to become degenerated. I saw a lot hardcode number inside GJK, is there anything i can fine tune?

thank you

**Comment by vreutskyy** [comment]
Hello. It would be great if you could make a compilable code snippet reproducing this issue. Thanks

**Reply by yananZZZZ** [reply]
Hi,
[SmallConvexShapeCannotbeTriggered.zip](https://github.com/NVIDIA-Omniverse/PhysX/files/12287001/SmallConvexShapeCannotbeTriggered.zip)

     Sorry for my last post didn't provide the code to duplicate the issue. Now I modified the demo project SnippetTrigger to make it clearer.
     What's I have done is I create a convex mesh and set it as trigger shape. My geometry is rather small so I also change some convex mesh description to make my hull not degenerated. The wire display shows my shape is ok.
     Next, I tried to create a free-fail box collision shape with similar dimension. then I found out that the "onTrigger" callback never triggered, So I change my free-fail object to capsule, the "onTrigger" is triggered, Can refer to the screenshot.
     What's more, I also tried to scale my convex mesh 100 times and make all object in the similar scale. the Trigger behavior is also normal.   So I highly suspect there is some tolerance issue related to the small box and convex gjk algorithm. 
     From the last post, I located that both scene query and trigger callback is actually using "intersectBoxConvex", it will be really great if someone can find out more about this strange behavior and if it indeed a tolerance issue maybe give some advise on how to handle this kind of small trigger as it widely used in my application.
    I also attached my code and it's created in V5.1.0. 
![1](https://g
![NoScaleBox](https://github.com/NVIDIA-Omniverse/PhysX/assets/122350362/2f5f3ac4-ffcb-4173-acf2-301f2753943b)
ithub.c
![NoScaleCapsule](https://github.com/NVIDIA-Omniverse/PhysX/assets/122350362/a2da8700-93f8-4e3e-ac75-273e0401dc9b)
om/NVIDIA-Omniverse/PhysX/assets/122350362/d9a82bf8-6d18-4588-bc11-922e441d55ed)
![ScaleBox](https://github.com/NVIDIA-Omniverse/PhysX/assets/122350362/11e2faee-9a37-43d2-b546-42d102399a7e)

Thank you.

**Reply by vreutskyy** [reply]
Sorry, I can't download your archive at
https://github.com/NVIDIA-Omniverse/PhysX/files/12287001/SmallConvexShapeCannotbeTriggered.zip

Could you check the link?

**Reply by yananZZZZ** [reply]
[](url)
[snippettriggers.zip](https://github.com/NVIDIA-Omniverse/PhysX/files/12321792/snippettriggers.zip)
How about this one?

**Reply by vreutskyy** [reply]
This one's okay, thanks 👍

**Reply by vreutskyy** [reply, marked as answer]
Hm. It's strangely similar to the neighboring issue here: https://github.com/NVIDIA-Omniverse/PhysX/discussions/173

Could you try the same fix I proposed them?

Change this line:
https://github.com/NVIDIA-Omniverse/PhysX/blob/e8c8deb2d548dc635db4f61083ea0e745c0102a0/physx/source/geomutils/src/gjk/GuGJKSimplex.h#L311

from
`                if(FAllGrtrOrEq(eps, area))`

to
`		if(FAllGrtrOrEq(FMul(eps, eps), area))`

Thanks

**Reply by yananZZZZ** [reply]
Hi,
After modify the code as suggested, the accuracy of small convex vs box is correct. Thank you for the support.
